### PR TITLE
Use SHA2.256 as default, add support for BLAKE2s, add spec for SHA1, Doc Updates

### PR DESCRIPTION
**Pull request by grempe**

### CODE REVIEW

File: GUIDE.md [hunk 1]
```diff
@@ -2,109 +2,86 @@
...
+```
 
 ## Verifing Fingerprints
 
 Files can be checked against a given fingerprint. In the case of `verify`, only changes and deletions will be reported. Additions are not reported.
 
 To generate a fingerprint for a given path:
 
-	fingerprint scan /bin > bin.fingerprint
+	fingerprint scan /tmp/test > /tmp/test.fingerprint
```
Comment (ioquatix) [inline GUIDE.md:56]: What was the reason for this change?

File: GUIDE.md [hunk 2]
```diff
@@ -254,4 +231,4 @@ Fingerprint can be used to produce printed documents which can be used to verify
 
 Simply follow the procedure to produce a cryptographic hash of a directory, print these documents out, and get them signed.
 
-N.B. Please consult a lawyer for the correct procedure and legal standing of such techniques. This document is not legal advise and we accepts no responsibility for any use or misuse of this tool.
+N.B. Please consult a lawyer for the correct procedure and legal standing of such techniques. This document is not legal advice and we accept no responsibility for any use or misuse of this tool.
```
Comment (ioquatix) [inline GUIDE.md:234]: Doh, thanks :)

File: lib/fingerprint/scanner.rb [hunk 3]
```diff
@@ -96,15 +98,21 @@ def digests_for(path)
 					@progress.call(total) if @progress
 					
 					@digests.each do |key, digest|
-						digest << buf
+						digest << buf unless key == 'BLAKE2s'
```
Comment (ioquatix) [inline lib/fingerprint/scanner.rb:101]: I am okay to merge this, but I'd like to add a layer of abstraction here to reduce the special casing.

File: lib/fingerprint/command/scan.rb [hunk 4]
```diff
@@ -37,7 +37,7 @@ class Scan < Samovar::Command
 				option "-p/--path <path>", "Analyze the given path relative to root.", default: "./"
 				
 				option "-x/--extended", "Include extended information about files and directories."
-				option "-s/--checksums <MD5,SHA1>", "Specify what checksum algorithms to use: #{CHECKSUMS.keys.join(', ')}.", default: DEFAULT_CHECKSUMS, type: Checksums
+				option "-s/--checksums <SHA2.256>", "Specify what checksum algorithms to use: #{CHECKSUMS.keys.join(', ')}.", default: DEFAULT_CHECKSUMS, type: Checksums
```
Comment (ioquatix) [inline lib/fingerprint/command/scan.rb:40]: We should probably include the comma to explain how to specify multiple checksums?

File: GUIDE.md (hunk 1 above)
Comment (grempe) [reply GUIDE.md:56]: To eliminate the need for the 'snip' where you truncated the fingerprint output, and to show examples that show the new SHA256 default, and not MD5.

File: lib/fingerprint/scanner.rb (hunk 3 above)
Comment (grempe) [reply lib/fingerprint/scanner.rb:101]: I'll leave that part to you :-) That's a non-trivial refactor.

File: lib/fingerprint/command/scan.rb (hunk 4 above)
Comment (grempe) [reply lib/fingerprint/command/scan.rb:40]: Doesn't that first arg in the angle brackets show what the defaults are? If you leave two entries in there with commas you are not really reflecting the default.  Not sure why defaulting to always using two different hashes would be beneficial.

**Comment by grempe** [comment]
Thanks for the merge!

**Pull request merged**